### PR TITLE
Specify event types for each event function

### PR DESCRIPTION
**Pull request by TrySound**

### CODE REVIEW

File: type-definitions/dom-event.d.ts [hunk 1]
```diff
@@ -2,29 +2,29 @@ import {Stream} from "most";
...
+export function keypress(node: EventTarget, capture?: boolean): Stream<KeyboardEvent>;
+export function keyup(node: EventTarget, capture?: boolean): Stream<KeyboardEvent>;
 export function input(node: EventTarget, capture?: boolean): Stream<Event>;
-export function contextmenu(node: EventTarget, capture?: boolean): Stream<Event>;
-export function resize(node: EventTarget, capture?: boolean): Stream<Event>;
-export function scroll(node: EventTarget, capture?: boolean): Stream<Event>;
+export function contextmenu(node: EventTarget, capture?: boolean): Stream<PointerEvent>;
+export function resize(node: EventTarget, capture?: boolean): Stream<UIEvent>;
+export function scroll(node: EventTarget, capture?: boolean): Stream<UIEvent>;
 export function error(node: EventTarget, capture?: boolean): Stream<Event>;
```
Comment (briancavalier) [inline type-definitions/dom-event.d.ts]: I wasn't quite sure where to look for these TS types, but google led me [heree](https://github.com/Microsoft/TypeScript/blob/master/src/lib/dom.generated.d.ts#L4209).  All of the occurrences of the `error` event I could find there seem to be of type `ErrorEvent`.  If that's right, then let's update it here, too.

**Review by briancavalier — CHANGES_REQUESTED** [review]
This is great, thanks!  One comment about also potentially updating the type of the "error" event.

File: type-definitions/dom-event.d.ts (hunk 1 above)
Comment (TrySound) [reply type-definitions/dom-event.d.ts]: Skipped. Done.

File: type-definitions/dom-event.d.ts [hunk 2]
```diff
@@ -25,10 +25,10 @@ export function input(node: EventTarget, capture?: boolean): Stream<Event>;
 export function contextmenu(node: EventTarget, capture?: boolean): Stream<PointerEvent>;
 export function resize(node: EventTarget, capture?: boolean): Stream<UIEvent>;
 export function scroll(node: EventTarget, capture?: boolean): Stream<UIEvent>;
-export function error(node: EventTarget, capture?: boolean): Stream<Event>;
+export function error(node: EventTarget, capture?: boolean): Stream<ErrorEvent>;
 
-export function hashchange(node: EventTarget, capture?: boolean): Stream<Event>;
-export function popstate(node: EventTarget, capture?: boolean): Stream<Event>;
+export function hashchange(node: EventTarget, capture?: boolean): Stream<HashChangeEvent>;
+export function popstate(node: EventTarget, capture?: boolean): Stream<PopStateEvent>;
```
Comment (briancavalier) [inline type-definitions/dom-event.d.ts:31]: :+1:

**Review by briancavalier — APPROVED** [review]
🚢

**Comment by briancavalier** [comment]
released in 1.4.0.  Thanks!

**Pull request merged**